src/java.base/share/classes/java/nio/X-Buffer.java.template

```diff
@@ -471,13 +471,16 @@ public abstract sealed class $Type$Buffer
      * Attempts to read characters into the specified character buffer.
      * The buffer is used as a repository of characters as-is: the only
      * changes made are the results of a put operation. No flipping or
-     * rewinding of the buffer is performed.
+     * rewinding of the buffer is performed. If the specified character
+     * buffer has no space {@linkplain Buffer#hasRemaining remaining},
+     * then no characters will be read and zero will be returned.
      *
      * @param target the buffer to read characters into
-     * @return The number of characters added to the buffer, or
-     *         -1 if this source of characters is at its end
+     * @return The number of characters added to the buffer,
+     *         possibly zero, or -1 if this source of characters is at its end
      * @throws IOException if an I/O error occurs
-     * @throws ReadOnlyBufferException if target is a read only buffer
+     * @throws ReadOnlyBufferException if target is a read only buffer,
+     *         even if it is empty
      * @since 1.5
```

Comment:
It might be clearer if this method linked to the length method. That way it could be specified to return 0 when the length of the target buffer is 0.

For the @throws of ROBE, probably need to avoid the word "empty", only because this method writes to the target buffer, it doesn't read from the target buffer.

Reply:
Addressed in 94505a45ce099faf3e31a07b8500ac456345a3b1; CSR not yet updated.

Reply:
I checked the update but it now has "no source remaining or its length is zero". In CB, length == remaining so I think you just want one in the description.

Reply:
Ah, good catch!

Reply:
Fixed by 7281b482c24a3c74da15dbf637f7627c5dc5407b.